modules/extensions/autoscaler.tf

```diff
@@ -118,7 +127,7 @@ data "helm_template" "cluster_autoscaler" {
 }
 
 resource "null_resource" "cluster_autoscaler" {
```

Comment:
I think it's cleaner to change this to `remote_cluster_autoscaler` so it matches the new `local_cluster_autoscaler`, however I didn't want to break anything upstream.  Options:

* Change the name and add a `moved {}` block.
* Leave the name as it is.

Any thoughts?

Reply:
In this case, then all extensions will need this kind of feature. I like your idea of using a variable to control remote or local exec. Perhaps just use the same null resource but different conditional blocks based on the variable?

Reply:
I ended up going with the `helm_release` resource, I think that is the better approach.  I attempted to use a separate null_resource with a local-exec provisioner, but I would also have to add a `local_file` resource to create the manifest and then apply it with kubectl.  Using kubectl makes sense when you can't use the `helm_release` resource remotely, but you can use it locally.

Reply:
I don't see the [helm provider](https://registry.terraform.io/providers/hashicorp/helm/latest/docs) being initialized.
I recommend using a [kube-config datasource](https://registry.terraform.io/providers/oracle/oci/latest/docs/data-sources/containerengine_cluster_kube_config) and creating a local kube-config file instead of relying on the existing one and the currently configured context.

Note that this approach assumes that OCI CLI is already installed and configured locally.